p2p/router.go

```diff
@@ -528,19 +590,25 @@ func (r *Router) dialPeers() {
 				return
 			}
 
-			queue := newPQScheduler(r.logger, peerID, r.metrics, r.chDescs, 0, 0, defaultCapacity)
-			queue.start()
+			var peerQueue queue
+			func() {
```

Comment:
Why put this into a function?

Reply:
to be able to use the defer on the unlock.

Reply:
we can avoid the defer by calling `r.peerMtx.Unlock()` directly after the code block. Note a blocker, I just find inline functions a bit harder to read.

Reply:
I moved this into it's own function, which I think improves readability. Given that the changes means that we have to call a function in the lock, I'm a bit afraid of doing that without a defer, as a matter of practice.